lib/ansible/modules/packaging/os/homebrew_cask.py

```diff
@@ -505,11 +517,28 @@ def _upgrade_all(self):
             self.message = 'Casks would be upgraded.'
             raise HomebrewCaskException(self.message)
 
-        rc, out, err = self.module.run_command([
-            self.brew_path,
-            'cask',
-            'upgrade',
-        ])
+        rc, out, err = '', '', ''
+
+        if self.sudo_password:
+            sudo_askpass_file = tempfile.NamedTemporaryFile()
+            sudo_askpass_file.write("#!/bin/sh\n\necho '" + self.sudo_password + "'\n")
+            os.chmod(sudo_askpass_file.name, 0o700)
```

Comment:
I'm no expert, but afaik ```tempfile.NamedTemporaryFile``` already creates the file with ```0600``` permission, not sure why you are setting this to ```0700```?

Reply:
SUDO_ASKPASS actually executes the file here, so we need execute permissions as well.